src/DotNet.Status.Web/TimelineIssueTriageService.cs

```diff
@@ -0,0 +1,286 @@
+
+                foreach (var comment in comments)
+                {
+                    // Look for category information in comment
+                    string category = GetTriageIssueProperty("category", comment.Body);
+                    if (!string.IsNullOrEmpty(category))
+                    {
+                        updatedCategory = category;
+                    }
+                }
+            }
+
+            foreach (var triageItem in triageItems)
+            {
+                triageItem.Url = issuePayload.Issue.Html_Url;
+                if (updatedCategory != null)
+                {
+                    triageItem.UpdatedCategory = updatedCategory;
+                    triageItem.Url = issuePayload.Issue.Html_Url;
+                }
+                _logger.LogInformation($"buildId: {triageItem.BuildId}, recordId: {triageItem.RecordId}, index: {triageItem.Index}, category: {triageItem.UpdatedCategory}, url: {triageItem.Url}");
+            }
+
+            await IngestTriageItemsIntoKusto(triageItems);
+
+            await gitHubClient.Issue.Comment.Create(issuePayload.Repository.Id, issuePayload.Issue.Number, $"Bot has updated the 'TimelineIssuesTriage' database.\n**PowerBI reports may take up to 24 hours to refresh**\n\nSee {_docLink} for more information.");
+
+            return;
+        }
+
+        private async Task IngestTriageItemsIntoKusto(ICollection<TriageItem> triageItems)
+        {
+            _logger.LogInformation("Entering IngestTriageItemIntoKusto");
+            await KustoHelpers.WriteDataToKustoInMemoryAsync(
+                _kustoIngestClientFactory.GetClient(),
+                _kustoOptions.Value.Database,
+                "TimelineIssuesTriage",
+                _logger,
+                triageItems,
+                b => new[]
+                {
+                    new KustoValue("ModifiedDateTime", b.ModifiedDateTime, KustoDataType.DateTime),
+                    new KustoValue("BuildId", b.BuildId, KustoDataType.Int),
+                    new KustoValue("RecordId", b.RecordId, KustoDataType.Guid),
+                    new KustoValue("Index", b.Index, KustoDataType.Int),
+                    new KustoValue("UpdatedCategory", b.UpdatedCategory, KustoDataType.String),
+                    new KustoValue("Url", b.Url, KustoDataType.String)
+                });
+        }
+
+        private bool IsDuplicate(IList<TriageItem> triageItems, IList<TriageItem> existingIssueItems)
+        {
+            return _logic.IsDuplicate(triageItems, existingIssueItems);
+        }
+
+        private string GetTriageIssueProperty(string propertyName, string body)
+        {
+            return _logic.GetTriageIssueProperty(propertyName, body);
+        }
+
+        private IList<TriageItem> GetTriageItemsProperties(string body)
+        {
+            return _logic.GetTriageItemsProperties(body);
+        }
+
+        public class TimelineIssueTriageLogic
```

Comment:
Logic is a bit vague tbh. I think the purpose of the class is to parse/scan text for triage information? Maybe something like `TriageInfoScanner`, `TriageInfoParser` or something like that would be more obvious?

Reply:
Also not sure how the rest of the project is but maybe put this in its own file?

Reply:
Unfortunately, most of the project, at least the status web, seems to have multiple classes related together in one file.

Reply:
The main reason to this service class is enhancing test-ability by the humble object pattern. I moved code which I consider fragile, essential and worth testing into public class.
I am not happy about it... as this class would have no rights to exists otherwise

Reply:
> Unfortunately, most of the project, at least the status web, seems to have multiple classes related together in one file.

Sure, then keep it as is. No worries!

> The main reason to this service class is enhancing test-ability by the humble object pattern. I moved code which I consider fragile, essential and worth testing into public class.

So the overall class doesn't seem that much complicated to be tested as a whole? Then we would have really nice tests for the full flow